src/engine/Filter.cpp

```diff
@@ -45,78 +44,104 @@ string Filter::getDescriptor() const {
   }
-  auto localVocab = subRes->getSharedLocalVocab();
-  return {[](auto subRes, auto* self) -> cppcoro::generator<IdTable> {
-            for (IdTable& idTable : subRes->idTables()) {
-              IdTable result = self->filterIdTable(subRes, idTable);
-              co_yield result;
-            }
-          }(std::move(subRes), this),
-          resultSortedOn(), std::move(localVocab)};
+  if (requestLaziness) {
+    return {[](auto subRes, auto* self) -> cppcoro::generator<IdTable> {
+              for (IdTable& idTable : subRes->idTables()) {
+                IdTable result = self->filterIdTable(subRes, idTable);
+                co_yield result;
+              }
+            }(std::move(subRes), this),
+            resultSortedOn(), std::move(localVocab)};
+  }
+
+  size_t width = getSubtree().get()->getResultWidth();
+  IdTable result =
+      ad_utility::callFixedSize(width, [this, &subRes, width]<int WIDTH>() {
+        IdTable result{width, getExecutionContext()->getAllocator()};
+        IdTableStatic<WIDTH> output =
+            std::move(result).toStatic<static_cast<size_t>(WIDTH)>();
+
+        for (IdTable& idTable : subRes->idTables()) {
+          computeFilterImpl(output, idTable, subRes->localVocab(),
+                            subRes->sortedBy());
+          checkCancellation();
+        }
+
+        return std::move(output).toDynamic();
+      });
+
+  LOG(DEBUG) << "Filter result computation done." << endl;
+
+  return {std::move(result), resultSortedOn(), std::move(localVocab)};
 }
 
 // _____________________________________________________________________________
 IdTable Filter::filterIdTable(const std::shared_ptr<const Result>& subRes,
-                              const IdTable& idTable) {
-  sparqlExpression::EvaluationContext evaluationContext(
-      *getExecutionContext(), _subtree->getVariableColumns(), idTable,
-      getExecutionContext()->getAllocator(), subRes->localVocab(),
-      cancellationHandle_, deadline_);
-
-  // TODO<joka921> This should be a mandatory argument to the
-  // EvaluationContext constructor.
-  evaluationContext._columnsByWhichResultIsSorted = subRes->sortedBy();
-
-  size_t width = evaluationContext._inputTable.numColumns();
-  IdTable result = CALL_FIXED_SIZE(width, &Filter::computeFilterImpl, this,
-                                   evaluationContext);
+                              const IdTable& idTable) const {
+  size_t width = idTable.numColumns();
+  IdTable result = ad_utility::callFixedSize(
+      width, [this, &idTable, width, &subRes]<int WIDTH>() {
+        IdTable result{width, getExecutionContext()->getAllocator()};
+        IdTableStatic<WIDTH> output =
+            std::move(result).toStatic<static_cast<size_t>(WIDTH)>();
+        computeFilterImpl(output, idTable, subRes->localVocab(),
+                          subRes->sortedBy());
+
+        return std::move(output).toDynamic();
+      });
```

Comment:
There is further room for getting rid of code duplication
(namely the `CALL_FIXED_SIZE` /moveToDynamic/moveToStatic stuff).

If you implement another helper, that has basically the same signature as `computeFilterImpl` (in particular the result is an output parameter), but the input and outputs are dynamic tables, and this function does exactly this call fixed size stuff, then you don't have the ugly `CALL_FIXED_SIZE` in two places + compile times go down + we can throw it out once we figure out whether it helps at all.